FIAT/hdiv_trace.py

```diff
@@ -49,47 +54,96 @@ class HDivTrace(FiniteElement):
     """
 
     def __init__(self, ref_el, degree):
+        """Constructor for the HDivTrace element.
+
+        :arg ref_el: A reference element, which may be a tensor product
+                     cell.
+        :arg degree: The degree of approximation. If on a tensor product
+                     cell, then provide a tuple of degrees if you want
+                     varying degrees.
+        """
         sd = ref_el.get_spatial_dimension()
         if sd in (0, 1):
-            raise ValueError("Cannot use this trace class on a %d-dimensional cell." % sd)
-
-        # Constructing facet element as a discontinuous Lagrange element
-        dglagrange = DiscontinuousLagrange(ufc_simplex(sd - 1), degree)
-
-        # Construct entity ids (assigning top. dim. and initializing as empty)
+            raise ValueError("Cannot take the trace of a %d-dim cell." % sd)
+
+        # Store the degrees if on a tensor product cell
+        if ref_el.get_shape() == TENSORPRODUCT:
+            try:
+                degree = tuple(degree)
+            except TypeError:
+                degree = (degree,) * len(ref_el.cells)
+
+            assert len(ref_el.cells) == len(degree), (
+                "Number of specified degrees must be equal "
+                "to the number of cells."
+            )
+        else:
+            if not ref_el.get_shape() in [TRIANGLE, TETRAHEDRON,
```

Comment:
I thought flake8 would complain for this. `x not in xs` is preferred over `not x in xs`.